assets/server/realmadmin/_stats_codes.html

```diff
@@ -212,12 +247,20 @@ <h5 class="modal-title">Mean time until code claimed</h5>
 
     let dataTable = new google.visualization.DataTable();
     dataTable.addColumn('date', 'Date');
-    dataTable.addColumn('number', 'Issued');
-    dataTable.addColumn('number', 'Claimed');
-    dataTable.addColumn('number', 'Invalid');
+    dataTable.addColumn('number', 'Codes Issued');
+    dataTable.addColumn('number', 'Codes Claimed');
+    dataTable.addColumn('number', 'Invalid Codes');
+    dataTable.addColumn('number', 'Tokens Claimed');
+    {{if .hasKeyServerStats}}
```

Comment:
Can we do this based on the JavaScript response instead?

Reply:
if we did, it would involve scanning every row and seeing if key server stats exist.
I think this is cleaner.

Reply:
Javascript is cheap, over a very small data set (N is usually < 100), and computed by the client instead of the server. Relying on information from the template makes it more difficult to move this into the application bundle in the future.

I also don't think you need to scan every row. I think you just need to scan the _last_ row. If there are key server stats in the last row, it means stats are enabled. If there aren't stats in the last row, it means that key server stats aren't enabled, _or_ they have recently been disabled. In either case, I think you arrive at the desired behavior. Am I missing something?

Reply:
or they haven't shown up yet for today...

Reply:
there's a race here and not all days will have key server status (min threshold)

Reply:
You could also export a `hasKeyServerStats` field in the _top-level_ JSON which mirrors the config for the realm on the server.